azure-mgmt-eventhub/tests/test_azure_mgmt_eventhub.py

```diff
@@ -0,0 +1,166 @@
+﻿# coding: utf-8
+
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+import unittest
+import time
+
+import azure.mgmt.eventhub.models
+from azure.mgmt.eventhub.models import EHNamespace
+from azure.mgmt.eventhub.models import EHNamespace, Sku, SkuName, AuthorizationRule, AccessRights, AccessKeys, Eventhub, CaptureDescription, Destination, EncodingCaptureDescription,  ErrorResponseException, ErrorResponse
+from azure.common.credentials import ServicePrincipalCredentials
+
+from devtools_testutils import AzureMgmtTestCase, ResourceGroupPreparer
+
+
+class MgmtEventHubTest(AzureMgmtTestCase):
+
+    def setUp(self):
+        super(MgmtEventHubTest, self).setUp()
+
+        self.eventhub_client = self.create_mgmt_client(
+            azure.mgmt.eventhub.EventHubManagementClient
+        )
+
+    @ResourceGroupPreparer()
+    def test_eh_eventhub_curd(self, resource_group, location):
+        # List all topic types
+        resource_group_name = resource_group.name #"ardsouza-resourcemovetest-group2"
+
+        # Create a Namespace
+        namespace_name = "testingpythontestcaseeventhubnamespaceEventhub"
+
+        namespaceparameter=EHNamespace(location,{'tag1':'value1','tag2':'value2'},Sku(SkuName.standard))
+        creatednamespace = self.eventhub_client.namespaces.create_or_update(resource_group_name, namespace_name, namespaceparameter, None, True).output
```

Comment:
Why are you doing that? create_or_update return a poller for you
intead of:
```python
creatednamespace = self.eventhub_client.namespaces.create_or_update(resource_group_name, namespace_name, namespaceparameter, None, True).output
self.assertEqual(creatednamespace.name, namespace_name)
while (self.eventhub_client.namespaces.get(resource_group_name,namespace_name).provisioning_state != 'Succeeded'):
    time.sleep(15)
    continue
```
this should be simply:
```python
poller = self.eventhub_client.namespaces.create_or_update(resource_group_name, namespace_name, namespaceparameter)
creatednamespace = poller.result()
self.assertEqual(creatednamespace.name, namespace_name)
```

I see this pattern a lot, please replace them all.

Reply:
one question here, as the namespace creation and deletion are LRO, so i have used raw=true. so used the .output. We need to verify that namespace is not in provisioning state before we proceed to entity creations, so keeping the while loop. please let me know if you any suggestions for this.

Reply:
@v-Ajnava The call to "result" is checking provisioningState for you